NetCasbin/InternalEnforcer.cs

```diff
@@ -17,34 +17,26 @@ public class InternalEnforcer : CoreEnforcer
         /// <returns></returns>
         protected Boolean AddPolicy(String sec, String ptype, List<String> rule)
         {
-            Boolean ruleAdded = model.AddPolicy(sec, ptype, rule);
-            if (!ruleAdded)
-            {
-                return false;
-            } 
-
             if (adapter != null && autoSave)
             {
                 try
                 {
                     adapter.AddPolicy(sec, ptype, rule);
                 }
-                catch (NotImplementedException e)
-                {
-
-                }
-                catch (Exception e)
-                {
-                    throw e;
-                }
-
-                if (watcher != null)
+                catch (NotImplementedException)
                 {
                     // error intentionally ignored
-                    watcher.Update();
                 }
             }
-            return true;
+
+            var ruleAdded = model.AddPolicy(sec, ptype, rule);
```

Comment:
I think this way won't have any effects. If `adapter.AddPolicy` failed, the model will still be updated, I have 2 ideas about this.

1. Modify `AddPolicy` to return a bool rather than void
2. If `AddPolicy` throws an exception on fail, `model.AddPolicy` shouldn't be executed.

I personally prefer method1, because if an adapter fails, we can do nothing to recover it, so throw an exception doesn't help much. Of course method 2 is also ok, as long as we keep model and adapter be consistent.

Reply:
I think also the method 1 better than method 2, but it may a breaking change to all adapter, If we use method 2 we can also catch all the exceptions to safely return. It like this:
```csharp
if (adapter != null && autoSave)
{
    try
    {
        adapter.AddPolicy(sec, ptype, rule);
    }
    catch (NotImplementedException)
    {
        // error intentionally ignored
    }
    catch
    {
        return false;
    }
}
```

Reply:
Agree with you, go ahead.

Reply:
throw exception is better here.

Reply:
> throw exception is better here.

I have thrown exception at here like before.